src/player/player_update.cpp

```diff
@@ -3117,11 +3120,14 @@ void UpdatePlayer()
 
 
                 // check vine backgrounds
-                for(B = 1; B <= numBackground; B++)
+                for(int B : treeBackgroundQuery(Player[A].Location, SORTMODE_NONE))
                 {
+                    if(B > numBackground)
+                        continue;
+
                     if(BackgroundFence[Background[B].Type] && (!g_compatibility.fix_climb_invisible_fences || !Background[B].Hidden))
                     {
-                        if(CheckCollision(Player[A].Location, Background[B].Location))
+                        // if(CheckCollision(Player[A].Location, Background[B].Location))
```

Comment:
Why did you remove the collision check with the vine BGOs? Are you absolutely sure player won't collide them outside the actual collision?

Reply:
I agree, that doesn't seem right. Let me look closer...

Reply:
Ah, right. Refer to the tempLocation collision below it, it was just a duplicated check that increased computational efficiency when collision was unlikely, but reduces computational efficiency when collision is likely.